photoview/build.gradle

```diff
@@ -16,24 +17,62 @@ dependencies {
     implementation "androidx.appcompat:appcompat:1.1.0"
 }
 
+def PUBLISH_GROUP_ID = 'com.github.chrisbanes'
+def PUBLISH_ARTIFACT_ID = 'PhotoView'
+def PUBLISH_VERSION = '2.3.0'
+
+ext["signing.keyId"] = ''
+ext["signing.password"] = ''
+ext["signing.secretKeyRingFile"] = ''
+ext["ossrhUsername"] = ''
+ext["ossrhPassword"] = ''
+ext["sonatypeStagingProfileId"] = ''
+
+File localPropsFile = project.rootProject.file('local.properties')
+if (localPropsFile.exists()) {
+    println "Found local props file, loading props from there"
+    Properties p = new Properties()
+    new FileInputStream(localPropsFile).withCloseable { is ->
+        p.load(is)
+    }
+    p.each { name, value ->
+        ext[name] = value
+    }
+} else {
+    println "No local props file, loading env vars instead"
+    ext["signing.keyId"] = System.getenv('SIGNING_KEY_ID')
+    ext["signing.password"] = System.getenv('SIGNING_PASSWORD')
+    ext["signing.secretKeyRingFile"] = System.getenv('SIGNING_SECRET_KEY_RING_FILE')
+    ext["ossrhUsername"] = System.getenv('OSSRH_USERNAME')
+    ext["ossrhPassword"] = System.getenv('OSSRH_PASSWORD')
+    ext["sonatypeStagingProfileId"] = System.getenv('SONATYPE_STAGING_PROFILE_ID')
+}
```

Comment:
- Set to empty values initially, so that they exist, but are empty. This allows people to build the debug variant locally.
- Use `local.properties` values if they're there, this can be used for publishing from a local machine.
- Use environment variables if the `local.properties` file was missing, can be used on CI.